x-pack/filebeat/module/fortinet/firewall/ingest/event.json

```diff
@@ -0,0 +1,518 @@
+{
+  "description": "Pipeline for parsing fortinet firewall logs (event pipeline)",
+  "processors": [
+    {
+      "set": {
+        "field": "event.kind",
+        "value": "event"
+      }
+    },
+    {
+      "set": {
+        "field": "event.outcome",
+        "value": "failure",
+        "if": "ctx.fortinet?.firewall?.result == 'ERROR'"
+      }
+    },
+    {
+      "append": {
+        "field": "event.type",
+        "value": [
+          "user",
+          "access",
+          "allowed",
+          "start"
+        ],
+        "if": "['FSSO-logon', 'auth-logon'].contains(ctx.fortinet?.firewall?.action)"
+      }
+    },
+    {
+      "append": {
+        "field": "event.type",
+        "value": [
+          "user",
+          "access",
+          "allowed",
+          "end"
+        ],
+        "if": "['FSSO-logoff', 'auth-logout'].contains(ctx.fortinet?.firewall?.action)"
```

Comment:
I think you want category authentication, type of end, & outcome success for a logout event, but maybe this represents more than that?

Reply:
This has been modified. The category is handled by checking for the "user" subtype a bit longer down the pipeline